x/btccheckpoint/README.md

```diff
@@ -0,0 +1,222 @@
+- [States](#states)
+  - [Parameters](#parameters)
+  - [Epoch data](#epoch-data)
+  - [Latest Finalized Epoch](#latest-finalized-epoch)
+  - [Submission data](#submission-data)
+  - [Transient States](#transient-states)
+- [Messages](#messages)
+  - [MsgInsertBTCSpvProof](#msginsertbtcspvproof)
+  - [MsgUpdateParams](#msgupdateparams)
+- [EndBlocker](#endblocker)
+- [Queries](#queries)
+
+## Concepts 
+Babylon's BTC Checkpoint module allows Babylon chain to periodically checkpoint its state onto the onto the Bitcoin blockchain whilst simultaneously verifying the checkpoints. The process involves two main components:
+
+Checkpoint Submission:
+1. When a new checkpoint for a specific epoch is available, the [vigilante](https://github.com/babylonlabs-io/vigilante) collects the necessary proof from the Bitcoin blockchain. 
+2. This proof is called an `SPVProof` (Simplified Payment Verification Proof), consisting of the Bitcoin transaction, Index of the transaction, the Merkle path and the Bitcoin header that confirms the transaction.
+3. The vigilante submits this proof to the Babylon chain using the `InsertBTCSpvProof` function. The proof is validated and stored in state. This includes parsing the submission, checking for duplicates, and verifying the checkpoint with the [checkpointing module](https://github.com/babylonlabs-io/babylon/blob/main/x/checkpointing/README.md).
+
+Checkpoint Verification:
+1. The Babylon chain maintains a Bitcoin light client through the [BTC Light Client module](https://github.com/babylonchain/babylon/blob/dev/x/btclightclient/README.md). This module is responsible for tracking Bitcoin block headers, allowing Babylon to verify the depth and validity of Bitcoin transactions without running a full Bitcoin node.
+2. When new Bitcoin blocks are produced, the headers are relayed to and processed by the Babylon chain's light client.
+3. As the light client's tip changes, it triggers the `OnTipChange` callback.
+4. This callback initiates the `checkCheckpoints` process,  which verifies the status of all submitted checkpoints based on their confirmation depth in the Bitcoin blockchain. This means that the Babylon chain will check if the checkpoint is still on the main chain, if the deepest (best) submission of older epoch happened before given submission, how deep is the submission, and mark each submission which is not known to btc light client or is on btc light fork as to delete. For more details on submissions see [Submissions](#submissions).
+5. Non-finalized epochs are retrieved from state. For each of these non-finalized epochs, the status is checked of the corresponding checkpoint. The depth of the checkpoint in the Bitcoin blockchain is verified and based on the depth and the module's parameters, the checkpoint's status may be updated. If the status changed, it's updated in the state and the corresponding status is set in the checkpointing module. Following the an epoch being finalized, all submissions are deleted.
```

Comment:
` Following the an epoch being finalized, all submissions are deleted` - > `Following the an epoch being finalized, all submissions except the best one are deleted`

After finalisation we leave the best submission in database.

Reply:
the an -> an